common/persistence/elasticsearch/visibility_store.go

```diff
@@ -719,24 +719,31 @@ func (s *visibilityStore) getNextPageToken(token []byte) (*visibilityPageToken,
 func (s *visibilityStore) getSearchResult(request *persistence.ListWorkflowExecutionsRequest, token *visibilityPageToken,
 	boolQuery *elastic.BoolQuery, overStartTime bool) (*elastic.SearchResult, error) {
 
-	matchNamespaceQuery := elastic.NewMatchQuery(searchattribute.NamespaceID, request.NamespaceID)
-	var rangeQuery *elastic.RangeQuery
-	if overStartTime {
-		rangeQuery = elastic.NewRangeQuery(searchattribute.StartTime)
-	} else {
-		rangeQuery = elastic.NewRangeQuery(searchattribute.CloseTime)
-	}
-
-	rangeQuery = rangeQuery.
-		Gte(request.EarliestStartTime).
-		Lte(request.LatestStartTime)
-
 	query := elastic.NewBoolQuery()
 	if boolQuery != nil {
 		*query = *boolQuery
 	}
 
-	query = query.Must(matchNamespaceQuery).Filter(rangeQuery)
+	matchNamespaceQuery := elastic.NewMatchQuery(searchattribute.NamespaceID, request.NamespaceID)
+	query = query.Must(matchNamespaceQuery)
+
+	if !request.EarliestStartTime.IsZero() && !request.LatestStartTime.IsZero() {
```

Comment:
Use ||, this block should run if either of the time is not zero.

Reply:
> Use force, Luke!

What a catch!